### ISSUE TITLE
Table 3. CPC comparision, ResNet 50 design choice

### ISSUE THREAD

**Issue by insikk** [rdevon/DIM#34]
Thank you for sharing your code. I'm happy to try this seemingly attractive representation learning idea on a different backbone. e.g. ResNet50.

Gladly, you mentioned ResNet50 on Table3 in your paper, so I'm looking into it. Then I got several questions about it.

Table3. CIFAR10 (no data augmentation) DIM(L) single global
![image](https://user-images.githubusercontent.com/2110359/65944341-74483180-e46c-11e9-8c95-02abee166606.png)
From the FoldedResNet code https://github.com/rdevon/DIM/blob/master/cortex_DIM/configs/resnets.py#L129 
I was able to reproduce similar result (conv: 80, Y(64): 71) as it is on the paper(80.95) with following commmand
```
CUDA_VISIBLE_DEVICES=0 python scripts/main.py local classifier --d.source CIFAR10 --d.batch_size 128 --o.learning_rate 0.0002 --encoder_config foldresnet34_32x32 --local.mode nce --local.mi_units 1024 -n CIFAR10_DIM_L_NCE_foldresnet34 --t.epochs 500
```
CIFAR10_DIM_L_NCE_foldresnet34_final.t7_global_test_acc
![image](https://user-images.githubusercontent.com/2110359/65945068-19173e80-e46e-11e9-862a-0b36a5eb5315.png)

I know you mentioned setting mi_units to be 2048 in Appendix 2, but it seems this change does not explain 9% performance difference. So I looked into the backbone. 

While I was trying to fill the performance gap, I found that the ResNet used in this code is somewhat different from the one I usually use. 

![image](https://user-images.githubusercontent.com/2110359/65943556-b40e1980-e46a-11e9-9696-52e5ecfdcf18.png)
From the above image (from original ResNet Paper), 34-layer and 50-layer structure are only differ in use of bottleneck block instead of basic block. 

In your code, even though the name shows resnet37, using bottleneck block make it resnet50 like encoder. 

And I also considered using ResNet for CIFAR10 requires small modification from ImageNet 224x224 input version of ResNet (e.g. First conv kernel size become 3x3 from 7x7, no maxpooling after that) 

However, there are a few more changes that have no explanation I can think of. Could you clarify what was the behind thought on these small changes?

* Why did you change all conv_2,3,4x layers to have stride=1? 
* The only downsampling occurs at the first conv. But you could do the same thing by setting stride at following conv layers. Is there any advantage downsampling on the lowest level feature?
* Why did you remove conv_5x layer completely from your _foldresnet37_32x32? I understand keeping all conv_xx layers make feature map to small (say, 1x1) but you can just set stride to be 1 as you did in previous layers. Does the insertion of conv_5x affect final results?
* repeating blocks follow [3, 4, 6, 3] pattern. Considering you removed the final block. Then following [3, 4, 6] seems reasonable. But your code has a pattern of [2, 5, 2]. Is there any specific reason?

I was tried various modified ResNet structure based on your code, and most of the time classification accuracy rise and down again, but at the same time loss still decreases. (maybe goes to a degenerative solution or trivial solution) Have you experienced this phenomenon during your work? If the above changes don't matter much to the final result I wouldn't ask this question. It seems DIM training works on specific design choice, and I'm not sure what it is. 

![image](https://user-images.githubusercontent.com/2110359/65944982-e2d9bf00-e46d-11e9-8bc1-3f5a4f98a55a.png)
![image](https://user-images.githubusercontent.com/2110359/65944965-d6edfd00-e46d-11e9-8093-e8a116b845c7.png)

**Comment by rdevon** [comment]
So one quick question: the results that you're focusing on are training DIM and the classifier together, correct?
